packet_manager.go

```diff
@@ -1,13 +1,19 @@
 package nex
 
+import "sync"
+
 // PacketManager implements an API for pushing/popping packets in the correct order
 type PacketManager struct {
 	currentSequenceID *Counter
 	packets           []PacketInterface
+	*sync.RWMutex
```

Comment:
I don't think we should export the mutex

```suggestion
	currentSequenceID *Counter
	packets           []PacketInterface
	mutex             *sync.RWMutex
```